src/main/java/org/opensearch/neuralsearch/query/NeuralSparseTwoPhaseParameters.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.opensearch.Version;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.ParsingException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.neuralsearch.settings.NeuralSearchSettings;
+import org.opensearch.neuralsearch.util.NeuralSearchClusterUtil;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Represents the parameters for neural_sparse two-phase process.
+ * This class encapsulates settings related to window size expansion, pruning ratio, and whether the two-phase search is enabled.
+ * It includes mechanisms to update settings from the cluster dynamically.
+ */
+@Getter
+@Setter
+@Accessors(chain = true, fluent = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class NeuralSparseTwoPhaseParameters implements Writeable {
+    public static volatile Float DEFAULT_WINDOW_SIZE_EXPANSION;
+    public static volatile Float DEFAULT_PRUNING_RATIO;
+    public static volatile Boolean DEFAULT_ENABLED;
+    public static volatile Integer MAX_WINDOW_SIZE;
+    @VisibleForTesting
+    static final ParseField NAME = new ParseField("two_phase_settings");
+    @VisibleForTesting
+    static final ParseField WINDOW_SIZE_EXPANSION = new ParseField("window_size_expansion");
+    @VisibleForTesting
+    static final ParseField PRUNING_RATIO = new ParseField("pruning_ratio");
+    @VisibleForTesting
+    static final ParseField ENABLED = new ParseField("enabled");
+    private static final Version MINIMAL_SUPPORTED_VERSION_TWO_PHASE_SEARCH = Version.CURRENT;
```

Comment:
Version.Current is not right way. Wondering how it worked for other features?

Reply:
Please take a look at what I respond you yesterday.

Reply:
https://github.com/opensearch-project/neural-search/pull/695#issuecomment-2076819753

Reply:
I checked that, but my question is simple. This is not the first feature who is facing this problem. So can you tell me how this particular use case is special

Reply:
The bwc framework is introduced after 2.11, and it's test cases(neural query, neural sparse query, and hybrid query) are already exists when we write this framework.

Reply:
> The bwc framework is introduced after 2.11, and it's test cases(neural query, neural sparse query, and hybrid query) are already exists when we write this framework.


So what difference does it make?

Reply:
Can you give an example of what you think the same case, and we take a look at whether they are difference.

Reply:
I'm not sure minimal supported version should be current. It mean with each release it will be incremented and not supported from previous release? Say we merge this before 2.15, then in 2.16 branch 2.16 is minimal supported meaning 2.15 is not supported